src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -224,18 +225,17 @@ public void systemDeletesCondition() {
 
 	@When("a user clicks on Attachments link from patient visits dashboard")
 	public void loadAttachmentsPage() {
-		visitsDashboardPage = (PatientVisitsDashboardPage) dashboardPage.goToRecentVisits();
-		attachmentsPage = (AttachmentsPage) dashboardPage.goToAttachmentsPage().waitForPage();
+		attachmentsPage = (AttachmentsPage) visitsDashboardPage.goToAttachmentsPage().waitForPage();
 	}
 
 	@Then("the system loads Attachments page")
 	public void systemloadsAttachmentsPage() {
-		assertTrue(textExists("Attachments"));
+		assertPage(attachmentsPage.waitForPage());
 	}
 
 	@When("a user attaches patient supporting file")
-	public void addSupportingFile() {
-		attachmentsPage.setFileUrl("/home/opensource/Documents/Form.pdf");
+	public void addSupportingFile() throws IOException {
+		attachmentsPage.setRandomFileUrl();
 		attachmentsPage.addAttachmentNote("Client medical history form");
 	}
 
```

Comment:
@kdaud I have created a dedicated method for handling the random file url which I have called here in the steps. Does this suffice?